docs: update greedy arrays note

mcecode opened the pull request:
As per https://github.com/yargs/yargs-parser/pull/515#issuecomment-3256844212, update the note to say that the possible change might be in v19 instead of v18.

mcecode committed:
docs: update greedy arrays note

shadowspawn reviewed (approved):
LGTM, thanks.

shadowspawn commented:
Changing the behaviour is not on my to-do currently. Do you want to update back to deleting the comment again? (I looked at reopening #515 but branch is gone. Sorry about the flip-flop.)

No worries if you have moved on in the meantime!

mcecode commented:
Hi, no worries, I understand that priorities change.

Sorry for deleting the branch, I have a habit of deleting old/unused branches right away.

To spare you another PR, I've just added a commit to this one and updated the title. Feel free to just squash it if ever so the `main` merge and my initial commit doesn't clutter history.

Thanks.

shadowspawn commented:
Thanks @mcecode
